notebooks/predicting_queries/entity_intent_suggestion.ipynb

```diff
@@ -0,0 +1,348 @@
+    "        hits.page.pageTitle as pageTitle,\n",
+    "        hits.page.pagePath as pagePath,\n",
+    "        TIMESTAMP_SECONDS(visitStartTime+CAST(hits.time/1000 AS INT64)) as pageview_timestamp,\n",
+    "        FROM\n",
+    "        `govuk-bigquery-analytics.87773428.ga_sessions_*` \n",
+    "        CROSS JOIN UNNEST(hits) AS hits\n",
+    "        WHERE _table_suffix BETWEEN FORMAT_DATE('%Y%m%d',DATE_SUB(CURRENT_DATE(), INTERVAL 2 DAY))\n",
+    "          AND FORMAT_DATE('%Y%m%d',DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY)) \n",
+    "        AND hits.type = \"PAGE\"\n",
+    "      ) as viewedpages\n",
+    "ON viewedpages.session_id = action.session_id\n",
+    "WHERE pageTitle is not null\n",
+    "AND pageview_timestamp < first_search_timestamp\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#queries_df = pandas_gbq.read_gbq(query)\n",
+    "#queries_df.to_csv(PROCESSED_DATA+'/pagehistoryqueries.csv')\n",
+    "queries_df = pd.read_csv(PROJECT_DIR+'/data/processed/pagehistoryqueries.csv')\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Query + session data\n",
+    "queries_df.head(10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Taxons for every bit of content \n",
+    "#taxons_df =graph.run(\"MATCH (c:Cid)-[r:IS_TAGGED_TO]->(t:Taxon)RETURN c.name as pagePath,t.name as taxon\").to_data_frame()\n",
+    "#taxons_df.to_csv(PROCESSED_DATA+'/taxons.csv')\n",
+    "taxons_df =pd.read_csv(PROJECT_DIR+'/data/processed/taxons.csv')\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# inner join taxons to session/query data \n",
+    "# We'll end up with multiple taxons per query \n",
+    "# ie each query has potentially many pages viewed prior to the query being made AND pages can have multiple taxons\n",
+    "\n",
+    "query_taxons = queries_df.merge(taxons_df,on='pagePath')\n",
+    "# Split queries up (they're stupidily aggregated in the sql)\n",
+    "query_taxons['query']=query_taxons['search_terms'].map(lambda x: x.split(','))\n",
+    "query_taxons = query_taxons.explode('query')\n",
+    "# Average number of taxons visited per query (around 7)\n",
+    "np.mean(query_taxons.groupby('session_id').size())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Instead of bag of words, let's try bag of entities\n",
+    "entities = pd.read_csv(PROJECT_DIR+'/data/processed/content_entities.csv')"
```

Comment:
Could just do the following instead since we have the `PROCESSED_DATA` variable.
```suggestion
    "entities = pd.read_csv(PROCESSED_DATA + '/content_entities.csv')"
```